### PR TITLE
Convert image and videos to blocks when switching from Classic editor to Block editor

### PR DESCRIPTION
**Pull request by dugajean**
Test notes:

- Have both Classic and Block editors ready for use
- Create a post using Classic Editor, insert image and/or video
- Save the post
- Switch to Gutenberg
- Image and video content should've been converted to their respective blocks rather than "Classic Block"

### CODE REVIEW

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php [hunk 1]
```diff
@@ -522,17 +522,99 @@ public function filter_video_embeds( $html, $id, $attachment ) {
...
+	 * 
+	 * @return string
+	 */
+	protected function classic_image_to_block( $content ) {
+		if ( strpos( $content, '<img class=' ) === false ) {
+			return $content;
+		}
+
+		$dom = new \DOMDocument();                
+		@$dom->loadHTML( $content, LIBXML_HTML_NODEFDTD );
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:581]: @dugajean 

I am wondering why we would want to silence the errors/warnings here.
Did you run into any while working on this?

Comment (dugajean) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:581]: `loadHTML` complains when the HTML you try to load isn't a full-fledged HTML document (ie. no <html>, <body>, etc.). So a way to remedy this is by silencing it as the warnings aren't really saying anything important.

Comment (dugajean) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:581]: More info here: https://stackoverflow.com/questions/9149180/domdocumentloadhtml-error

Comment (pereirinha) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:584]: thanks for the explanation and comment in the code.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php [hunk 2]
```diff
@@ -522,17 +522,102 @@ public function filter_video_embeds( $html, $id, $attachment ) {
...
+			$img_clone->setAttribute( 'class', 'wp-image-' . $img_id );
+
+			// Prep the <figure> node
+			$img_fig = $dom->createElement( 'figure' );
+			$img_fig->appendChild( $img_clone );
+			$img_fig->setAttribute( 'class', 'wp-block-image' );
+			
+			// Surround <figure> with block comments
+			$fragment = $dom->createDocumentFragment();
+			$fragment->appendChild( $dom->createComment( ' wp:image {"id":' . $img_id . ',"sizeSlug":"large"} ' ) );
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:608]: @dugajean 

We should try to read the image size from the `$img` element — from the class name, eg. `size-thumbnail` —, and not assuming that we'll always use the `large` size.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php [hunk 3]
```diff
@@ -522,17 +522,147 @@ public function filter_video_embeds( $html, $id, $attachment ) {
...
+		// Fetch image id
+		preg_match( '/wp-image-([0-9]+)/', $class_list, $matches );
+		$results[] = $id = isset( $matches[1] ) ? (int) $matches[1] : null;
+
+		// Fetch image size
+		$results[] = $size = 'large';
+		if ( false !== strpos( $class_list, 'size-' ) ) {
+			preg_match( '/size-([A-Za-z0-9-_]+)/', $class_list, $matches );
+
+			$all_sizes = get_intermediate_image_sizes();
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:643]: @dugajean 

What you're trying to do here is great and it makes sense. However, this function on VIP will return you an [empty array](https://github.com/WordPress/WordPress-Coding-Standards/commit/e7707b2338e7823a2d14cfbdca8bdc88488d6081#diff-1034de57626216004cbb6c6e1707f313R244-R250).

I'd say to be ok to implement the same logic of the [function](https://developer.wordpress.org/reference/functions/get_intermediate_image_sizes/), as `wp_get_additional_image_sizes()` seems to be a safe method.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php [hunk 4]
```diff
@@ -522,17 +522,147 @@ public function filter_video_embeds( $html, $id, $attachment ) {
...
+
+		// Fetch image alignment
+		$results[] = $align = null;
+		if ( false !== strpos( $class_list, 'align' ) && false === strpos( $class_list, 'alignnone' ) ) {
+			preg_match( '/align([a-z]+)/', $class_list, $matches );
+			$results[2] = $align = isset( $matches[1] ) ? $matches[1] : null;
+		}
+
+		// Prep json for block comment
+		$results[] = json_encode(
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:655]: Can we use here `wp_json_encode()` instead?

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php [hunk 5]
```diff
@@ -522,17 +522,147 @@ public function filter_video_embeds( $html, $id, $attachment ) {
...
+		$results[] = $align = null;
+		if ( false !== strpos( $class_list, 'align' ) && false === strpos( $class_list, 'alignnone' ) ) {
+			preg_match( '/align([a-z]+)/', $class_list, $matches );
+			$results[2] = $align = isset( $matches[1] ) ? $matches[1] : null;
+		}
+
+		// Prep json for block comment
+		$results[] = json_encode( 
+			array(
+				'id' 	   => $id,
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php:657]: There are here missed spaces and tabs in this indentation.

**Pull request closed without merging**